ferry.go

```diff
@@ -93,8 +97,71 @@ func (f *Ferry) newDataIterator() (*DataIterator, error) {
 		dataIterator.CursorConfig.BuildSelect = f.CopyFilter.BuildSelect
 	}
 
-	err := dataIterator.Initialize()
-	return dataIterator, err
+	return dataIterator
+}
+
+func (f *Ferry) NewDataIteratorWithoutStateTracker() *DataIterator {
+	dataIterator := f.NewDataIterator()
+	dataIterator.StateTracker = nil
+	return dataIterator
+}
+
+func (f *Ferry) NewBinlogStreamer() *BinlogStreamer {
+	return &BinlogStreamer{
+		DB:           f.SourceDB,
+		DBConfig:     f.Source,
+		MyServerId:   f.Config.MyServerId,
+		ErrorHandler: f.ErrorHandler,
+		Filter:       f.CopyFilter,
+	}
+}
```

Comment:
I'm doing this because nothing external to Ferry is creating its own DataIterator/BinlogStreamer instances. I tried to have the alternative, but ended up initializing DataIterator/BinlogStreamer (for the delta copier and the reconciler) by copying all the attributes needed directly from Ferry. 

I figured it may be better to turn the Ferry into a factory. I feel like this line of thinking is more natural anyway as we have a global config object passed to Ferry, and Ferry is already the God object managing all the states.